WISE-Community/WISE
# Add snackbar success and error notifications for a variety of http requests

Added snackbar success and error notifications to share run and share project dialogs as well as student and teacher edit profile and edit password pages.  I also added a generic http error interceptor catches all HttpErrorResponse erroer and shows a snackbar notification alerting user. Closes #1504.

(Also improved unit tests for each of these  and the copy project and create run dialogs.)

To test:
1. Log in as a teacher and share a run with another teacher, edit shared teacher permissions, and remove a shared teacher. You should see a snackbar notifying you of each change.
2. Select the Browse WISE Units tab.
3. Share a project with another teacher, edit shared teacher permissions, and remove a shared teacher. You should see a snackbar notifying you of each change.
4. Go to `/teacher/profile/edit` and make some changes to profile. Change password. You should see snackbar notifications for all updates.
5. Log in as a student and go to `/student/profile/edit` and make some changes to profile. Change password. You should see snackbar notifications for all updates.
6. Try performing any of the above actions with your browsers Network set to 'offline'. You should see a snackbar alerting you of an error and asking you to check your network.
